### PR TITLE
feature/integration-with-datadog

### PR DESCRIPTION
**Pull request by yandongxiao**
Signed-off-by: yandongxiao <yandongxiao@starrocks.com>

### CODE REVIEW

File: helm-charts/charts/kube-starrocks/templates/starrocks/starrockscluster.yaml [hunk 1]
```diff
@@ -177,23 +197,23 @@ spec:
...
 {{ toYaml .Values.starrocksBeSpec.nodeSelector | indent 6 }}
 {{- end }}
-{{- if or .Values.starrocksBeSpec.beEnvVars .Values.initPassword.enabled }}
     beEnvVars:
-    {{- if .Values.timeZone }}
       - name: TZ
         value: {{ .Values.timeZone }}
-    {{- end }}
-    {{- if and .Values.initPassword.enabled (.Values.starrocksBeSpec.beEnvVars | toJson | contains "MYSQL_PWD" | not) }}
+      {{- if .Values.datadog.log.enabled }}
```
Comment (kevincai) [inline helm-charts/charts/kube-starrocks/templates/starrocks/starrockscluster.yaml:203]: is it possible that there is no env var to be rended at last, and the crd will be invalid. like
```
beEnvVars:
other configure: xxx

File: doc/intergrations/intergration-with-datadog.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+**Add** the following configuration to your `sr-values.yaml` file:
+
+```yaml
+datadog:
+  log:
+    enabled: true   # enable the log collection
+    tags: '["env:test"]'
+  metrics:
+    enabled: false  # enable the metrics collection
```
Comment (kevincai) [inline doc/intergrations/intergration-with-datadog.md:60]:
```suggestion
    enabled: true  # enable the metrics collection
```

File: doc/intergrations/intergration-with-datadog.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    annotations:
+      prometheus.io/path: "/metrics"
+      prometheus.io/port: "8040"
+      prometheus.io/scrape: "true"
+  beEnvVars:
+    - name: LOG_CONSOLE
+      value: "1"
+```
+
+So, if you want to get more control of the log and metric configuration, you can modify configuration on starrocks
```
Comment (kevincai) [inline doc/intergrations/intergration-with-datadog.md:104]: this is ambiguous, sounds like the user can use `kubectl` to modify the starrocks cluster. I suppose not.

File: helm-charts/charts/kube-starrocks/templates/starrocks/starrockscluster.yaml (hunk 1 above)
Comment (yandongxiao) [reply helm-charts/charts/kube-starrocks/templates/starrocks/starrockscluster.yaml:203]: now, beEnvVars will always has `TZ`.

Comment (yandongxiao) [reply helm-charts/charts/kube-starrocks/templates/starrocks/starrockscluster.yaml:203]: If timeZone is not set , or timeZone=="" in values.yaml, I think it equals set TZ environemnt to "" in container, and also that is the same as not set TZ environment.

pods are all ready:
```
 k get pods
NAME                                      READY   STATUS    RESTARTS   AGE
kube-starrocks-be-0                       1/1     Running   0          98s
kube-starrocks-fe-0                       1/1     Running   0          3m3s
kube-starrocks-operator-b758f6d47-mh2r5   1/1     Running   0          5m27s
```

in pod, like this:
```
env:
    - name: TZ
    - name: POD_NAME
      valueFrom:
        fieldRef:
          fieldPath: metadata.name
```
in container, like this:
```
root@kube-starrocks-fe-0:/opt/starrocks# env | grep TZ
TZ=
root@kube-starrocks-fe-0:/opt/starrocks# date
Mon Jul 31 09:51:07 UTC 2023.   # UTC time
```

**Pull request merged**